### PR TITLE
Creating lobby API config and making the UUID customizable

### PR DESCRIPTION
**Pull request by flamecoals**
Improving lobby, as discussed in #354 ...
We need this change to make it possible to use the Lobby API in [FreeBoardGame.org](https://freeboardgame.org). In FBG users send a link to invite friends to a match, and the current UUID is too long and too verbose for a link.

#### Checklist

* [x] Use a separate branch in your local repo (not `master`).
* [x] Test coverage is 100% (or you have a story for why it's ok).

### CODE REVIEW

**Comment by nicolodavis** [comment]
@flamecoals Do you think it makes sense to just make the default a shorter uuid that is suitable for your purpose?

This PR is fine too, but just curious.

**Comment by flamecoals** [comment]
I think it does make sense, yes, I think making it easy to share a URL could be a built in feature on the framework.

**Comment by flamecoals** [comment]
I can change the default library to be [shortid](https://www.npmjs.com/package/shortid) ... That's the one we currently use. Do you want me to do that, @nicolodavis ?

**Comment by nicolodavis** [comment]
Yeah, let's go with shortid as the default.

File: src/server/api.js [hunk 1]
```diff
@@ -57,12 +63,15 @@ export const CreateGame = async (db, game, numPlayers, setupData) => {
...
+export const createApiServer = ({ db, games, lobbyConfig }) => {
   const app = new Koa();
-  return addApiToServer({ app, db, games });
+  return addApiToServer({ app, db, games, lobbyConfig });
 };
 
-export const addApiToServer = ({ app, db, games }) => {
+export const addApiToServer = ({ app, db, games, lobbyConfig }) => {
+  if (!lobbyConfig || !lobbyConfig.shortid) {
+    lobbyConfig = { ...lobbyConfig, shortid };
```
Comment (nicolodavis) [inline src/server/api.js]: If lobbyConfig is `undefined` or `null`, then wouldn't this object-spread cause an error?  I think you want something like:

```
if (!lobbyConfig) {
  lobbyConfig = {};
}

if (!lobbyConfig.shortid) {
  lobbyConfig = { ...lobbyConfig, shortid };
}
```

File: src/server/api.js [hunk 2]
```diff
@@ -44,11 +50,11 @@ export const CreateGame = async (db, game, numPlayers, setupData) => {
   });
 
   for (let playerIndex = 0; playerIndex < numPlayers; playerIndex++) {
-    const credentials = createCredentials();
+    const credentials = lobbyConfig.shortid();
```
Comment (nicolodavis) [inline src/server/api.js]: Shall we call this option something more generic like `uuid`?  If someone overrides it to generate a longer ID, then `shortid` won't be a particularly relevant name.

File: src/server/api.js [hunk 3]
```diff
@@ -33,8 +31,16 @@ const GameMetadataKey = gameID => `${gameID}:metadata`;
  * @param {number} numPlayers - The number of players.
  * @param {object} setupData - User-defined object that's available
  *                             during game setup.
+ * @param {object } serverConfig - Configuration options for the server.
```
Comment (nicolodavis) [inline src/server/api.js]: Looks like this can be removed.

**Pull request merged**